crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space/bootstrap_task.rs

```diff
@@ -0,0 +1,708 @@
+        bootstrap_net: BootstrapNet,
+        bootstrap_check_delay_backoff_multiplier: u32,
+    ) -> Arc<RwLock<Self>> {
+        let this = Arc::new(RwLock::new(BootstrapTask {
+            is_finished: false,
+            current_delay: Duration::from_secs(1),
+            max_delay: Duration::from_secs(60 * 60),
+        }));
+        let bootstrap_query = DefaultBootstrapService {
+            url: bootstrap_service,
+            net: bootstrap_net,
+        };
+
+        let task_this = this.clone();
+        BootstrapTask::spawn_inner(
+            task_this,
+            internal_sender,
+            host_sender,
+            space,
+            Box::new(bootstrap_query),
+            bootstrap_check_delay_backoff_multiplier,
+        )
+    }
+
+    fn spawn_inner(
+        this: Arc<RwLock<Self>>,
+        internal_sender: GhostSender<SpaceInternal>,
+        host_sender: Sender<KitsuneP2pEvent>,
+        space: Arc<KitsuneSpace>,
+        bootstrap_query: Box<impl BootstrapService + Send + Sync + 'static>,
+        bootstrap_check_delay_backoff_multiplier: u32,
+    ) -> Arc<RwLock<Self>> {
+        let task_this = this.clone();
+        tokio::spawn(async move {
+            let backoff_multiplier = if bootstrap_check_delay_backoff_multiplier < 2 {
+                tracing::warn!(
+                    "Using default bootstrap backoff multiplier 2 because configured value is too low - {}",
+                    bootstrap_check_delay_backoff_multiplier
+                );
+                2
+            } else {
+                bootstrap_check_delay_backoff_multiplier
+            };
+
+            let max_delay = task_this.read().max_delay;
+
+            loop {
+                if !internal_sender.ghost_actor_is_active() {
+                    break;
+                }
+
+                let current_delay = task_this.read().current_delay;
+                tokio::time::sleep(current_delay).await;
+                if current_delay <= max_delay {
+                    // Backoff but don't exceed the configured max delay
+                    task_this.write().current_delay =
+                        std::cmp::min(current_delay * backoff_multiplier, max_delay);
+                }
+
+                match bootstrap_query
+                    .random(RandomQuery {
+                        space: space.clone(),
+                        limit: 8.into(),
+                    })
+                    .await
+                {
+                    Err(e) => {
+                        tracing::error!(msg = "Failed to get peers from bootstrap", ?e);
+                    }
+                    Ok(list) => {
+                        if !internal_sender.ghost_actor_is_active() {
+                            break;
+                        }
+                        let mut peer_data = Vec::with_capacity(list.len());
+                        for item in list {
+                            // TODO - @neonphog someday some validation here
```

Comment:
@neonphog do you know if the validation that should be done here is tracked as an issue?

Reply:
Mmm. The validation we are going to want to do is against DeepKey, which will make more sense to do on the holochain side, as there are multiple sources of peer data (such as new connection preflights) that we'd like to validate all in a DRY manner. So I think you can probably delete this TODO here.

Reply:
Thanks for the explanation, I'll remove this as suggested